mattermost/mattermost
# [MM-55270] fix(accessibility): "channels" button in global header is not selectable with keyboard

#### Summary


The product branding only handled click events to open the switcher menu. These changes make it possible to click on the product branding using <kbd>Space</kbd> and <kbd>Enter</kbd> to open/close the switcher menu.

#### Ticket Link


Fixes https://github.com/mattermost/mattermost/issues/27068
Jira https://mattermost.atlassian.net/browse/MM-55270

#### Screenshots


<details><summary>Before</summary>

https://github.com/user-attachments/assets/518f5032-ce81-40b2-907f-1b484f1c54bd

</details> 

<details><summary>After</summary>

https://github.com/user-attachments/assets/3698f795-7d8a-4c49-979f-7bdc2745b35c

</details> 

#### QA Review Test Steps

1. Click the global header's empty portion to the search bar's left.
2. Press <kbd>Tab</kbd> until the product branding i.e. "Channels" is focused.
3. Press <kbd>Space</kbd> and verify it opens the switcher menu.
4. Press <kbd>Space</kbd> again and verify it closes the switcher menu.
5. Repeat steps 3 and 4 with the <kbd>Enter</kbd> key.
6. Repeat step 1 to remove focus and then left-click the product branding twice.
7. Verify this action toggles the switcher menu.

#### Release Note

```release-note
* Added the ability to toggle the switcher menu in the global header using the SPACE 
and ENTER keys while the product branding is in focus.
```
